### PR TITLE
Privacy link footer option

### PR DESCRIPTION
**Pull request by jmsharp-utah**

### CODE REVIEW

File: @utahdts/utah-design-system-header/src/@types/jsDocTypes.d.js [hunk 1]
```diff
@@ -268,6 +268,8 @@
  * @typedef FooterSettings {
  *  @property {string | null | undefined} [copyrightYear] - the year to show next to the copy right symbol in the footer (default is none)
  *  @property {DomLocationTarget} [domLocationTarget] - where in the DOM should the footer be inserted? (defaults to the bottom of the body)
+ *  @property {string | null | undefined } [linkPrivacyPolicy] - a custom link to a privacy policy
```
Comment (AustinHaws-Utah) [inline @utahdts/utah-design-system-header/src/@types/jsDocTypes.d.js:271]: `| undefined` and [linkPrivacyPolicy] both mean undefined. is this intentional? 
[linkPrivacyPolicy] means it does not need specified
`| undefined` means you can pass in undefined
`| undefined [linkPrivacyPolicy]` means it doesn't need passed in and if you do pass it in you can set it to undefined.
it's tempting to suggest just having `| null` and not `| undefined`.

File: @utahdts/utah-design-system-header/src/js/renderables/footer/html/Footer.html [hunk 2]
```diff
@@ -24,10 +24,10 @@
         <a href="https://www.utah.gov/index.html" target="__blank" rel="noreferrer">Utah.gov Home</a>
       </li>
       <li>
-        <a href="https://www.utah.gov/support/disclaimer.html" target="__blank" rel="noreferrer">Terms of Use</a>
+        <a href="https://www.utah.gov/support/disclaimer.html" target="__blank" rel="noreferrer" id="utds-footer-terms-link">Terms of Use</a>
```
Comment (AustinHaws-Utah) [inline @utahdts/utah-design-system-header/src/js/renderables/footer/html/Footer.html:27]: long line? might be better with split attributes on separate lines?

Comment (AustinHaws-Utah) [reply @utahdts/utah-design-system-header/src/js/renderables/footer/html/Footer.html:27]: oh, this is an html file. i get in trouble for saying this about html code.

File: @utahdts/utah-design-system-header/src/js/renderables/footer/renderFooter.js [hunk 3]
```diff
@@ -102,6 +102,20 @@ export default function renderFooter() {
         footerHr?.remove();
       }
 
+      // add custom privacy policy link
+      if (settings?.footer?.linkPrivacyPolicy) {
+        const privacyLink = document.getElementById(domConstants.FOOTER_LINK_PRIVACY_ID);
+        checkForError(!privacyLink, 'renderFooter: cannot find privacy policy link');
+        privacyLink?.setAttribute('href', settings.footer.linkPrivacyPolicy);
```
Comment (AustinHaws-Utah) [inline @utahdts/utah-design-system-header/src/js/renderables/footer/renderFooter.js:109]: is it possible the href was already set? some of the other onClick and other handlers check to make sure something wasn't already set before setting and throw an error if so... probably not pertinent here.

File: @utahdts/utah-design-system-header/src/@types/jsDocTypes.d.js (hunk 1 above)
Comment (jmsharp-utah) [reply @utahdts/utah-design-system-header/src/@types/jsDocTypes.d.js:271]: change to: `{string | null}`

**Pull request merged**